mail_parent_recipient/models/mail_template.py

```diff
@@ -0,0 +1,45 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Camptocamp
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import models, api
+from odoo.tools import config
+
+
+class MailTemplate(models.Model):
+
+    _inherit = 'mail.template'
+
+    @api.multi
+    def generate_recipients(self, results, res_ids):
+        """Use partner's parent email as recipient.
+
+        Walk up the hierarchy of recipient partners via `parent_id`
+        and pick the 1st one having an email.
+        """
+        results = super(MailTemplate, self).generate_recipients(
+            results,
+            res_ids
+        )
+        test_condition = (
+            config['test_enable'] and not self.env.context.get(
+                'test_parent_mail_recipient'
```

Comment:
why do we need this flag?

Reply:
It is to enable the machinery of this module during test ONLY when the 'test_parent_mail_recipient'  is set.

Reply:
This is not answering the main question: why do we need this flag?
I guess is to not make other tests fail.
IMO this should be configured either at company level (off by default) or at partner level (off by default).
We can start adding a flag at company level 1st + res.config.settings then we'll see.